### PR TITLE
[DataPack] Fix few memory leaks and a few more race conditions

### PR DESCRIPTION
**Pull request by Bara**
This is definitely a few breaking changes, if something went wrong. So please test, before review.

### CODE REVIEW

File: addons/sourcemod/scripting/surftimer/hooks.sp [hunk 1]
```diff
@@ -1457,23 +1457,27 @@ public Action ResetOneJump(Handle timer, any userid)
 public Action DelayedVelocityCap(Handle timer, Handle pack)
 {
 	ResetPack(pack);
-	int client = ReadPackCell(pack);
+	int client = GetClientOfUserId(ReadPackCell(pack));
 	float speedCap = ReadPackFloat(pack);
+	delete pack;
```
Comment (PwnK) [inline addons/sourcemod/scripting/surftimer/hooks.sp:1462]: DelayedVelocityCap is a timer callback which is called by [`CreateDataTimer`](https://sm.alliedmods.net/new-api/timers/CreateDataTimer) so we don't need to close the datapack which comes as it's parameter.

Before making changes that close datapacks in timer callback parameters I recommend to always check if it's called by [`CreateTimer`](https://sm.alliedmods.net/new-api/timers/CreateTimer) or [`CreateDataTimer`](https://sm.alliedmods.net/new-api/timers/CreateDataTimer).

Comment (Bara) [reply addons/sourcemod/scripting/surftimer/hooks.sp:1462]: Yeah I remember... 🤦‍♂️
https://github.com/surftimer/Surftimer-olokos/pull/103#pullrequestreview-407890736

I changed that to CreateTimer now.

**Comment by olokos** [comment]
I am not really sure why change all CreateDataTimer to CreateTimer though.
https://github.com/alliedmodders/sourcemod/issues/682

Given above issue, CreateDataTimer would be the preffered method to call, there's less complexity later when those methods are changed/removed as you don't have to manually close all handles, it's done automatically with CreateDataTimer.

Correct me if I'm wrong here, but I see no upsides of switching to CreateTimer

**Comment by olokos** [comment]
Merged, but with this change 
e868c8ec9530e082849a891ae75e5b0356d91b87
delete handles were reintroduced, do we really need them if we're only using CreateDataTimer?
@Bara 
Let me know your take on this, should I create an issue so we can tackle this later on?

**Pull request merged**